plugin.php

```diff
@@ -17,36 +17,159 @@
 
 namespace Alley\WP\REST_API_Guard;
 
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Server;
+
 if ( ! defined( 'ABSPATH' ) ) {
 	exit;
 }
 
-// Check if Composer is installed.
-if ( ! file_exists( __DIR__ . '/vendor/autoload.php' ) ) {
-	\add_action(
-		'admin_notices',
-		function() {
-			?>
-			<div class="notice notice-error">
-				<p><?php esc_html_e( 'Composer is not installed and rest-api-guard cannot load. Try using a `*-built` branch if the plugin is being loaded as a submodule.', 'plugin_domain' ); ?></p>
-			</div>
-			<?php
-		}
-	);
+/**
+ * Instantiate the plugin.
+ */
+function main() {
+	require_once __DIR__ . '/settings.php';
 
-	return;
+	add_action( 'rest_pre_dispatch', __NAMESPACE__ . '\on_rest_pre_dispatch', 10, 3 );
```

Comment:
`rest_pre_dispatch` is a filter, not an action:
```suggestion
	add_filter( 'rest_pre_dispatch', __NAMESPACE__ . '\on_rest_pre_dispatch', 10, 3 );
```